ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java

```diff
@@ -1751,15 +1751,22 @@ public Table getTable(final String dbName, final String tableName, String metaTa
     }
 
     Table t = new Table(tTable);
-    if (metaTableName != null) {
-      if (t.getStorageHandler() == null || !t.getStorageHandler().isMetadataTableSupported()) {
-        throw new SemanticException(ErrorMsg.METADATA_TABLE_NOT_SUPPORTED, t.getTableName());
+    if (tableIdentifier != null) {
+      if (t.getStorageHandler() == null || !t.getStorageHandler().isTableIdentifierSupported()) {
+        throw new SemanticException(ErrorMsg.TABLE_IDENTIFIER_NOT_SUPPORTED, t.getTableName());
       }
-      if (!t.getStorageHandler().isValidMetadataTable(metaTableName)) {
-        throw new SemanticException(ErrorMsg.INVALID_METADATA_TABLE_NAME, metaTableName);
+      if (tableIdentifier.startsWith("branch_")) {
```

Comment:
please move string literal "branch_" to constant

Reply:
Fixed. I use the added util method `HiveUtils::getTableBranch` to avoid this.